### PR TITLE
.*: support binlog and pocket workload

### PR DESCRIPTION
**Pull request by you06**
### What problem does this PR solve? 

Support binlog in k8s and pocket workload

### What is changed and how does it work?

Add a control mode which does not rely on the default scheduler which invoke every round.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by you06** [comment]
This PR is ready but should merge #40 and #41 first, because the interface change will make the above two PRs not work. I should make some changes after #40 and #41 got merged.

**Comment by zhouqiang-cl** [comment]
@you06 please solve conflict

**Comment by you06** [comment]
@mahjonp @mapleFU PTAL

File: cmd/tidb/main.go [hunk 1]
```diff
@@ -102,6 +105,10 @@ func main() {
 		log.Fatalf("invalid checker %s", *checkerNames)
 	}
 
+	if withProf {
+		cfg.Mode = control.ModeWithPerf
+	}
+
```
Comment (you06) [inline cmd/tidb/main.go]: Set `Mode` field to `ModeWithPerf` to enable perf.

File: cmd/util/suit.go [hunk 2]
```diff
@@ -79,6 +79,7 @@ func (suit *Suit) Run(ctx context.Context) {
 	if len(suit.Config.ClientNodes) == 0 {
 		log.Panic("no client nodes exist")
 	}
+	suit.Config.ClientCount = len(suit.Config.ClientNodes)
```
Comment (mahjonp) [inline cmd/util/suit.go]: Why need this?

File: db/tidb/bank.go [hunk 3]
```diff
@@ -203,12 +203,16 @@ func (c *bankClient) DumpState(ctx context.Context) (interface{}, error) {
 	return balances, nil
 }
 
+func (c *bankClient) Start(ctx context.Context, cfg interface{}, clientNodes []clusterTypes.ClientNode) error {
+	return nil
```
Comment (mahjonp) [inline db/tidb/bank.go]:
```suggestion
	panic("unreachable")
```

File: pkg/test-infra/pkg/tidb/operation.go [hunk 4]
```diff
@@ -447,3 +522,41 @@ func getDiscoveryMeta(tc *v1alpha1.TidbCluster) (metav1.ObjectMeta, label.Label)
...
+
+func (t *TidbOps) parseNodeFromPodList(pods *corev1.PodList) []clusterTypes.Node {
+	var nodes []clusterTypes.Node
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.ObjectMeta.Name, "discovery") {
+			continue
+		}
+		nodes = append(nodes, clusterTypes.Node{
+			Namespace: pod.ObjectMeta.Namespace,
+			PodName:   pod.ObjectMeta.Name,
```
Comment (mahjonp) [inline pkg/test-infra/pkg/tidb/operation.go]: Need fill the `Component` field here,

https://github.com/pingcap/tipocket/blob/6dba244c5f1eed302f5e2928463eb90ad1051d09/pkg/cluster/k8s.go#L115-L129

File: pkg/test-infra/pkg/tidb/recommend.go [hunk 5]
```diff
@@ -18,15 +18,18 @@ import (
...
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type TiDBClusterRecommendation struct {
 	*v1alpha1.TidbCluster
 	*corev1.Service
+	NS   string
+	Name string
```
Comment (mahjonp) [inline pkg/test-infra/pkg/tidb/recommend.go]: NS and Name are same as TidbCluster.ObjectMeta.Namespace and TidbCluster.ObjectMeta.Name, so I think we can remove these two new fields. What's your opinion?

File: cmd/util/suit.go (hunk 2 above)
Comment (you06) [reply cmd/util/suit.go]: Maybe we should remove this field

File: pkg/test-infra/pkg/tidb/recommend.go (hunk 5 above)
Comment (you06) [reply pkg/test-infra/pkg/tidb/recommend.go]: In binlog cluster, two tidb cluster with different names are in the same namespace.

https://github.com/pingcap/tipocket/blob/005f3f1824075707165fbc32dcf1da671c2f4f79/pkg/test-infra/pkg/binlog/recommander.go#L49-L50

**Pull request merged**